flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorTest.java

```diff
@@ -2896,7 +2899,9 @@ private TaskExecutorTestingContext createTaskExecutorTestingContext(
                                 .setTaskStateManager(stateStoresManager)
                                 .setTaskChangelogStoragesManager(changelogStoragesManager)
                                 .build(),
-                        HEARTBEAT_SERVICES,
+                        // prevent heartbeat timeouts from failing the tests focused on other
+                        // aspects
+                        HeartbeatServices.noOp(),
```

Comment:
Did we check that none of the other tests are affected by this change? Besides `TaskExecutorTest.testSharedResourcesLifecycle`, also `testDynamicSlotAllocation`, `testTaskSlotTableTerminationOnShutdown` and `testReleasingJobResources` are using this utility method and this configuration. I'm wondering whether we should make this a method parameter and just use it for `testSharedResourcesLifecycle` to be on the safe side?

Reply:
Done in 9a33560928fb01405164b13023e874a6a8e6fa66.